pkgs/top-level/all-packages.nix

```diff
@@ -4691,7 +4691,7 @@ let
   sdcc = callPackage ../development/compilers/sdcc { };
 
   smlnjBootstrap = callPackage ../development/compilers/smlnj/bootstrap.nix { };
-  smlnj = callPackage_i686 ../development/compilers/smlnj { };
+  smlnj = callPackage ../development/compilers/smlnj { };
```

Comment:
@acowley: I'm not sure if this is a good idea.  From what I've gathered, callPackage_i686 is needed for for using 32-bit binaries on 64-bit systems.  See: https://botbot.me/freenode/nixos/msg/44602266/

Perhaps we should have a separate expression for Darwin which could replace @jwiegley's `smlnjBootstrap`...

cc @thoughtpolice